openfl/interface/envoy.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Collaborator manager CLI."""
+
+import logging
+import shutil
+import sys
+from importlib import import_module
+from os import path
+from pathlib import Path
+
+import click
+from click import group
+from click import option
+from click import pass_context
+from click import Path as ClickPath
+from yaml import safe_load
+
+from openfl.component.envoy.envoy import Envoy
+from openfl.interface.cli_helper import SITEPACKS
+from openfl.interface.cli_helper import WORKSPACE
+
+logger = logging.getLogger(__name__)
+
+
+@group()
+@pass_context
+def envoy(context):
+    """Manage Federated Learning Envoy."""
+    context.obj['group'] = 'collaborator-manager'
+
+
+@envoy.command(name='start')
+@option('-n', '--shard-name', required=True,
+        help='Current shard name')
+@option('-d', '--director-uri', required=True,
+        help='The FQDN of the federation director')
+@option('--disable-tls', default=False)
+@option('-sc', '--shard-config-path', default='shard_config.yaml',
+        help='The shard config path', type=ClickPath(exists=True))
+@option('-rc', '--root-cert-path', 'root_ca', default=None,
+        help='Path to a root CA cert')
+@option('-pk', '--private-key-path', 'key', default=None,
+        help='Path to a private key')
+@option('-oc', '--public-cert-path', 'cert', default=None,
+        help='Path to a signed certificate')
+def start_(shard_name, director_uri, disable_tls, shard_config_path,
+           root_ca, key, cert):
+    """Start the collaborator manager."""
+    logger.info('🧿 Starting the Collaborator Manager.')
+
+    shard_descriptor = shard_descriptor_from_config(shard_config_path)
+    envoy = Envoy(
+        shard_name=shard_name,
+        director_uri=director_uri,
+        shard_descriptor=shard_descriptor,
+        disable_tls=disable_tls,
+        root_ca=root_ca,
+        key=key,
+        cert=cert
+    )
+
+    envoy.start()
+
+
+@envoy.command(name='create-workspace')
+@option('-p', '--collaborator-manager-path', required=True,
+        help='The Collaborator Manager path', type=ClickPath())
+def create(envoy_path):
```

Comment:
Should be `collaborator-manager-path` or  `('-p', '--collaborator-manager-path', 'envoy_path'...` in 67 line

Reply:
Changed